app.py

```diff
@@ -164,13 +200,27 @@ def generate_response(prompt):
     with response_container:
         for i in range(len(st.session_state['generated'])):
             message = st.session_state['generated'][i]
-            st.markdown(f"<div class='you'>🧑‍🎓: {st.session_state['past'][i]}</div>", unsafe_allow_html=True)
-            st.markdown(f"<div class='waam'>🏫: {message}</div>", unsafe_allow_html=True)
-            if message not in st.session_state.votes:
-                st.session_state.votes[message] = [0, 0]
-            upvote_button = st.button(f"👍 ({st.session_state.votes[message][0]})", key=f"upvote_{i}")
-            downvote_button = st.button(f"👎 ({st.session_state.votes[message][1]})", key=f"downvote_{i}")
+            question = st.session_state['past'][i]
+            st.markdown(f"<div class='you'>🧑‍🎓: {question}</div>", unsafe_allow_html=True)
+            for m in message:
+                print(m[:6])
+                if m[:6] == 'figure':
+                    print('DEBUG: EXECING FIGURE')
+                    # exec and render
+                    figure = m[6:]
+                    try:
+                        exec(figure)
+                        st.write(ret)
```

Comment:
`ret` must be cleared after it is rendered or else it will be re rendered every time ```figure stuff happens but there is no new graph, some kid is trying do a code injection. Such an edge case lol.
Just add 
```suggestion
                        st.write(ret)
                        ret = None
```